mlflow.yaml

```diff
@@ -140,6 +146,77 @@ subpackages:
             run-script --help
         - uses: test/tw/ldd-check
 
+  - name: mlflow-iamguarded-compat
+    description: MLflow with iamguarded additions
+    options:
+      no-depends: true
+      no-provides: true
+    dependencies:
+      provides:
+        - mlflow=${{package.full-version}}
+      runtime:
+        - bash
+        - busybox
+        - git
+        - libpq
+        - merged-usrsbin
+        - py3-pip
+        - python3
+        - wolfi-baselayout
+    pipeline:
+      - uses: iamguarded/build-compat
+        with:
+          package: ${{package.name}}
+          version: ${{vars.major-version}}
+      - runs: |
+          # Restore virtual env paths
+          sed -i "s|/usr/share/mlflow|/home/build/venv|g" venv/pyvenv.cfg
+          sed -i "s|/usr/share/mlflow|/home/build/venv|g" venv/bin/*
+
+          # Re-initialize and activate virtual env
+          python -m venv venv --system-site-packages
+          source venv/bin/activate
+
+          # Install additional dependencies used by iamguarded
+          pip install -I --no-compile annotated_types anyio boto3 botocore build distlib \
+            distro exceptiongroup filelock h11 httpcore httpx jmespath openai platformdirs \
+            prometheus_client prometheus_flask_exporter psycopg2 pydantic pydantic_core \
+            pyproject_hooks s3transfer sniffio tomli tqdm virtualenv wheel
+
+          # Upgrade requests to fix GHSA-9hjg-9r4m-mvj7
+          pip install --upgrade "requests>=2.32.4"
+
+          # Remove pip
+          pip uninstall --yes pip
+
+          # Use Python in virtual environment
+          sed -i "s|/home/build/venv|/usr/share/mlflow|g" venv/pyvenv.cfg
+          sed -i "s|/home/build/venv|/usr/share/mlflow|g" venv/bin/*
+
+          # Install virtual environment
+          mkdir -p ${{targets.contextdir}}/usr/share/mlflow
+          cp -r venv/* ${{targets.contextdir}}/usr/share/mlflow/
+
+          # Create auth symlink expected in iamguarded Helm chart
+          mkdir -p ${{targets.contextdir}}/iamguarded
```

Comment:
Based on what I've learnt from @wojciechka we don't need/want the `${{targets.contextdir}}` here because the next pipeline handles it.

```suggestion
          mkdir -p /iamguarded
```